packages/core/src/components/overflow-list/overflowList.tsx

```diff
@@ -156,8 +156,14 @@ export class OverflowList<T> extends React.PureComponent<IOverflowListProps<T>,
         }
     }
 
+    public shouldComponentUpdate(_nextProps: IOverflowListProps<T>, nextState: IOverflowListState<T>) {
+        return !(this.state !== nextState && shallowCompareKeys(this.state, nextState));
```

Comment:
seems suspect to only check state and not include a props-changed check here.

Reply:
@invliD can you comment on why this works without the props check? probably worth leaving code comment to document _why_ we have to write a custom `sCU` here (since that's a very rare occurrence).

Reply:
that's not quite what this is doing. It means "always update, unless state has changed identity but is still the same (all keys shallow equal the previous value)". This is what caused double-renders in React 15.

Reply:
ah perfect. please add `// comment` so we don't forget.